File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeclareTypesInNamespaces.cs
using System;
using System.Collections.Generic;

public class Foo // Noncompliant {{Move 'Foo' into a named namespace.}}
{
    class InnerFoo // Compliant - we want to report only on the outer class
    {
    }
}

public struct Bar // Noncompliant {{Move 'Bar' into a named namespace.}}
{
    struct InnerBar // Compliant - we want to report only on the outer struct
    {
    }

    enum InnerEnu // Compliant - we want to report only on outer enum
    {
    }
}

public enum Enu // Noncompliant {{Move 'Enu' into a named namespace.}}
{
    Test
}

namespace Tests.Diagnostics
{
    class Program { }

    struct MyStruct { }

    public interface MyInt { }

    public enum Enu
    {
        Test
    }
}

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeclareTypesInNamespaces.vb
Imports System
Imports System.Collections.Generic

Public Class Foo ' Noncompliant {{Move the type 'Foo' into a named namespace.}}
    Class InnerFoo ' Compliant - we want to report only on the outer class
    End Class
End Class

Public Structure Bar ' Noncompliant {{Move the type 'Bar' into a named namespace.}}
    Structure InnerBar ' Compliant - we want to report only on the outer struct
    End Structure
End Structure

Namespace Tests.Diagnostics
    Class Program
    End Class

    Structure MyStruct
    End Structure
End Namespace
